scripts/build.py

```diff
@@ -100,13 +115,95 @@ def copy_files_source_to_target(source, target):
                 continue
             if source not in source_path:
                 continue
-            target_path = source_path.replace(
-                source, target)
+
+            target_path = source_path
+            relative_filepath = os.path.relpath(source_path, source)
+            if (not relative_filepath.endswith(FILE_EXTENSIONS_NOT_TO_RENAME) or
+                    relative_filepath.endswith(FILE_ENDINGS_TO_RENAME)):
+                filepath, file_extension = os.path.splitext(source_path)
+                target_path = (filepath + '.' +
```

Comment:
Looks like you do this sort of construction in multiple places; I suggest pulling out a private function to handle it.